The Big Guide to Web Components Built With React/The Big Guide to Web Components Built With React.md

```diff
@@ -0,0 +1,434 @@
+
+This portability and isolation can enable many of the benefits of Micro Frontends, which is why we decided to pursue Web Components as a Micro Frontends solution. [We'll discuss this at length later.](#web-components-as-micro-frontends)
+
+But if we're using React, isn't some of that portability and isolation already built in?
+
+## Why Build Web Components With React? Why not just _use React_?
+
+In a Web Component, you're not limited in the technology you use, so any approach should be feasible, but if you're using a technology like React already, why not simply ship shared React components for different pieces of functionality?
+
+There are at a few answers to this question: 
+
+1. Shared components within the React system depend on a common set of React functionality. If you're building a component, and you want to use a newer feature, but the global version of React in your system does not support this feature, then you simply can't use it. A Web Component has no need to have a top-down version of React dictating the features it can use. The Web Component can use the version it needs.
+
+2. Evolvability: the isolation of Web Components allows to you use React or really any other technology you see fit. If your team decides that something better or more fitting has come along for your Web Component, you can change it without worrying about affecting the surrounding page. Or you can build a Web Component without any third party library at all, and it can integrate in any web page easily (even those built with React).
+
+3. Possibly the most important answer is deployability. One of the main concerns that a Micro Frontends approach aims to alleviate is a lack of dependency on external teams when deploying an application. If you're using React to piece together your frontend, all of your components need to be up to date at build time. With a Web Component whose source is hosted on an independent server, you can deploy new source code (which we [go into detail about](#file-naming-cache-busting-and-deployment) below), and your downstream dependents will be updated automatically.
+
+There are some limitations and hurdles when it comes to using React in Web Components, and some questions we wanted to answer as we approached the idea  Let's talk about those. The first is the problem of lost functionality.
+
+## The Functionality Problem
+
+The [React Docs describe a way you can use React to render Web Components](https://reactjs.org/docs/web-components.html#using-react-in-your-web-components), but a deeper exploration of this idea reveals an issue: anything that uses events within your React Web Component to update the internal state simply does not work. You can [see an example of this broken functionality on this Sandbox](https://codesandbox.io/s/react-web-component-without-retargeting-events-b61u3?file=/src/index.js), (displayed below):
+
+<iframe
```

Comment:
This iframe isn't rendered in the output.  The markdown processor turns `src` into an `a` and mangles the iframe tag.  Maybe because of some line breaks in here.

Reply:
I think it's just because Github's Markdown intentionally doesn't render iframes. I can remove this if we want to keep this hosted on Github. (https://stackoverflow.com/questions/48402823/embed-openstreetmap-iframe-in-github-markdown)